pkg/componentreadiness/jiraautomator/jiraautomator.go

```diff
@@ -0,0 +1,597 @@
+			if len(col.RegressedTests) > 0 {
+				columnKeyBytes, err := json.Marshal(col.ColumnIdentification)
+				if err != nil {
+					return componentRegressedTests, err
+				}
+				columnID := string(columnKeyBytes)
+				columnToRegressionCount[columnID]++
+
+				// find all the defined variantBasedComponentRegressionThreshold this column is relevant to
+				for k, n := range col.Variants {
+					v := Variant{Name: k, Value: n}
+					if threshold, ok := j.variantBasedComponentRegressionThreshold[v]; ok {
+						columnToVariantsToThreshold[columnID] = map[Variant]int{v: threshold}
+					}
+				}
+			}
+		}
+	}
+	for _, row := range report.Rows {
+		for _, col := range row.Columns {
+			columnKeyBytes, err := json.Marshal(col.ColumnIdentification)
+			if err != nil {
+				return componentRegressedTests, err
+			}
+			columnID := string(columnKeyBytes)
+			useVariantBasedComponent := false
+			if variantToThreshold, ok := columnToVariantsToThreshold[columnID]; ok {
+				for variant, threshold := range variantToThreshold {
+					if threshold < columnToRegressionCount[columnID] {
+						if component, ok := variantBasedComponents[variant]; ok {
+							componentRegressedTests[component] = append(componentRegressedTests[component], col.RegressedTests...)
+							useVariantBasedComponent = true
+						}
+					}
+				}
+			}
+			if !useVariantBasedComponent {
+				componentRegressedTests[row.Component] = append(componentRegressedTests[row.Component], col.RegressedTests...)
+			}
+		}
+	}
+	for component, tests := range componentRegressedTests {
+		sort.Slice(tests, func(i, j int) bool {
+			return tests[i].ReportStatus < tests[j].ReportStatus
+		})
+		componentRegressedTests[component] = tests
+	}
+	return componentRegressedTests, nil
+}
+
+func (j JiraAutomator) integrateJiraForView(view crtype.View) error {
+
+	logger := log.WithField("view", view.Name)
+	logger.Info("jira integration for view")
+
+	report, err := j.getComponentReportForView(view)
+	if err != nil {
+		logger.WithError(err).Error("error getting report for view")
+	}
+
+	componentRegressedTests, err := j.groupRegressedTestsByComponents(report)
+	if err != nil {
+		logger.WithError(err).Error("error getting regressed tests from report")
+	}
+	for component, tests := range componentRegressedTests {
+		if component != "Test Framework" {
+			continue
+		}
+		// fetch jira bugs
+		// resetting this to our component to test
+		component = "Test Framework"
+		if j.componentWhiteList.Len() > 0 && !j.componentWhiteList.Has(component) {
+			continue
+		}
+		issues, err := j.getExistingIssuesForComponent(view, component)
+		if err != nil {
+			log.WithError(err).Error("error getting existing jira issues")
+		}
+
+		// No existing issues, create new one
+		if len(issues) == 0 {
+			_, err := j.createNewJiraIssueForRegressions(view, component, tests, nil)
+			if err != nil {
+				log.WithError(err).Error("error creating jira issue")
+			}
+		} else {
+			selected := issues[0]
```

Comment:
The contents of this else look intensive, might be good to break out an updateJiraForRegressions to match the above.

Reply:
WIll do.